### ISSUE TITLE
General assistance to start container 

### ISSUE THREAD

**Issue by whi4** [dperson/openvpn-client#123]
Hi,
I am very new to using docker on a bananapi running armbian and omv.  I am finding it a little difficult to startup this container for PIA.
I use:
docker run -it --cap-add=NET_ADMIN --device /dev/net/tun --name vpn             -v /srv/dev-disk-by-id-ata-ST2000LM007-1R8174_WDZ8VQM6-part1/shares/transmission:/vpn -d dperson/openvpn-client             -v 'Singapore;realuserID;realpassword'

The container exits, and I see:
root@bpip:~# docker logs 0f614b706ca1
standard_init_linux.go:190: exec user process caused "exec format error"

I have had a look at the Dockerfile, and the openvpn.sh file.  I was hoping to see very informative comments in the log to explain why it exited.  

I am really new to docker, and finding that alot of images for armhf do not start on my machine.  

I have googled and it appears that "exec format error" is pretty serious.

What sort of things should I do to rectify this situation?   Any help would be greatly appreciated.

Regards,  Robert

**Comment by whi4** [comment]
Hi again,
I rebuilt the docker image and now have a working armhf version.  (Not sure what I did wrong before.)

Anyway, I am now having problems getting a working vpn connection.  I cannot understand the instructions on starting up a PIA connection. I have downloaded the file https://www.privateinternetaccess.com/openvpn/openvpn.zip.

I have copied the certificate file called ca.rsa.2048.crt  to vpn-ca.crt in my shared volume.

I want to connect to the Singapore server, so I looked up its url in the Singapore.ovpn file to use in the docker startup, 

docker run -it --cap-add=NET_ADMIN --device /dev/net/tun --name vpn \
            -v /srv/dev-diskxxx/shares/transmission:/vpn -d armhf-openvpn-client \
            -v 'sg.privateinternetaccess.com;puser;wpass'

The docker starts up and seems to run, with the following message:

Wed Jun 13 14:51:50 2018 OpenSSL: error:14007086:SSL routines:CONNECT_CR_CERT:certificate verify failed
Wed Jun 13 14:51:50 2018 TLS_ERROR: BIO read tls_read_plaintext error
Wed Jun 13 14:51:50 2018 TLS Error: TLS object -> incoming plaintext read error
Wed Jun 13 14:51:50 2018 TLS Error: TLS handshake failed
Wed Jun 13 14:51:50 2018 SIGUSR1[soft,tls-error] received, process restarting

So I might be nearly there, but not quite!!

Should I be specifying the ovpn file somewhere?  I cannot work out what I am doing wrong.  

Regards,  Robert

**Comment by whi4** [comment]
Hi,
I got it working for PIA.  I am listing my startup in case it helps someone else:

docker run -it --cap-add=NET_ADMIN --device /dev/net/tun --name vpn \
            -e "CIPHER=aes-128-cbc" \
            -e "AUTH=sha1" \
            -v /srv/dev-disk/shares/transmission:/vpn -d armhf-openvpn-client \
            -v 'sg.privateinternetaccess.com;username;password;1198'

I found that basically the openvpn.sh script rewrites the .ovpn file and I have used the available environment variables and other stuff to ensure that vpn.conf has the same contents.  

The last thing I tried was to use port 1198.  It seems that PIA offers different "features" on different ports.

Anyway, for the moment I am happy.

Regards,  Robert
